---
title: "ALL_SWITCHES"
author: "ES"
date: "2025-12-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
```{r}
# Load libraries
library(forcats)
library(Polychrome)
library(ggplot2)
library(gghighlight)
library(ggpubr)
library(extrafont)
library(tidyr)
library(dplyr)
library(ggbreak)
library(scales)
library(ggallin)
library(stringr)
library(MsCoreUtils)
library(imputeLCMD)
library(tidytable)
```

# STEP 1 of workflow - take raw mass spectrometry files and process them using DIA-NN. Align all samples to all switched proteomes of interest as well as the reference

# STEP 2 of workflow - using the pr matrices generated by DIA-NN, filter rows for only proteotypic valeus in Perseus and save as REF/SWITCH_proteotypic.txt and convert to csv

# STEP 3 of workflow - impute missing values using imputeLCMD in both the REF and SWITCh proteotypic csv files- MinDet method assumes that values for peptides have not been detected due to low abundance, therefore takes the minimum value detected in each sample and assigns this to all missing values.
#Example uses W>F, repeat for all switches of interest
```{r}
reference <-read.csv("U:\\switched_proteomes\\total_proteome\\REF\\step2\\switch_proteotypic.csv")
switch <- read.csv("U:\\switched_proteomes\\total_proteome\\W-X\\W-F\\step2\\switch_proteotypic.csv")

reference <- as.matrix(reference)
switch <- as.matrix(switch)

# Select columns with abundance values - for DIA-NN dataset with 12 samples, this is the first 12
switch_filtered <-switch[,c(1:12)]
reference_filtered <- reference[,c(1:12]

# Define for imputeLCMD which columns to impute
cols_to_impute_switch <- colnames(switch_filtered)
class(switch_filtered) <- "numeric"

# Impute missing values                   
imputed_values <- impute.MinDet(switch_filtered, q = 0.01)
                                   
# Rejoin imputed values with original data
switch[, colnames(imputed_values)] <- imputed_values

#Repeat for reference data
cols_to_impute_reference <- colnames(reference_filtered)
class(switch_filtered) <- "numeric"

# Impute missing values                   
imputed_values <- impute.MinDet(reference_filtered, q = 0.01)
                                   
# Rejoin imputed values with original data
reference[, colnames(imputed_values)] <- imputed_values

# Export data
write.csv(switch, file = "U:\\switched_proteomes\\total_proteome\\W-X\\W-F\\step3\\SWITCH_report.pr_matrix_imputed.csv")
write.csv(reference, file = "U:\\switched_proteomes\\total_proteome\\REF\\step3\\REF_report.pr_matrix_imputed.csv")

```
# STEP 4 of workflow - in Perseus load in the reference and switched imputed data. For ease, filter REF for original amino acid of interest (e.g. W) and filter SWITCH for chosen amino acid of interest (e.g.F). Join dataframes via the Multi-proc. Matching rows by name functionality and export as tsv
# In Excel, convert tsv into csv generate matches data by first seperating out Stripped.Sequences in Data/Text to Column functionality using ';' as seperator.
# using COUNTIF formula, determine if the switched equivalent of a given peptide is also present in the reference proteome
# Save as matches_data.txt

# Back in Perseus, rejoin both the REF and SWITCH imputed data with matches_data. 
# Export as txt and save as REF/SWITCH_report.pr_matrix_with_matches.csv
                                   
# STEP 6 - Taking REF_with_matches txt converted to csv from Perseus, you can plot differential peptide expression between conditions of interest (for me, this is Parental Cell line treated with 10Gy versus KO cell line treated with 10Gy) to see upregulated peptides that also have a switch

```{r}
# load data
    pr_log2c_wt10g_ko10g <- read.csv("U:\\switched_proteomes\\total_proteome\\W-X\\W-F\\results\\step5\\REF_report.pr_matrix_log2_wt10g_ko10g.csv")

# format so multiple entries of matches only appear as 1

pr_log2c_wt10g_ko10g$Matches.Count <- sub("^([0-9.]+);.*", "\\1", pr_log2c_wt10g_ko10g$Matches.Count)
pr_log2c_wt10g_ko10g$Matches.Count[pr_log2c_wt10g_ko10g$Matches.Count == "NaN"] <- 0
```

```{r}

# Highlight only peptides with a switch in the switched proteome


plot <- pr_log2c_wt10g_ko10g %>% ggplot(aes(x=Difference,y=X.NAME.)) +
  geom_point()+ geom_hline(yintercept = 1.3, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed") + geom_vline(xintercept = 1, linetype="dashed")+
    gghighlight(Matches.Count > 0) +
  geom_text(
    data = pr_log2c_wt10g_ko10g %>% filter(Matches.Count >0),
    aes(label = Genes),
    position = position_jitter(width = 0.2, height = 0.25),
    color = "blue")+ 
    theme(axis.text = element_text(size = 10),
        plot.title = element_text(hjust = 0.5),
       legend.position = "none",
       panel.border = element_rect(color="gray3", fill= NA),
       panel.background=element_blank(),
        text=element_text(size = 12,family = "Helvetica"))+ xlab("log2 fold change")+
  ylab("-log p") +ggtitle("K>V peptides present in Parental 10Gy versus KO 10Gy")
  
```

# export

```{r}
ggsave(plot, file= "U:\\switched_proteomes\\total_proteome\\W-X\\W-F\\results\\step5\\pr_log2FC_WT10G_KO10G.pdf", height = 7, width = 7, scale=1, device=cairo_pdf)
```

# load in perseus data


## reference with matches data

```{r}
  ref <- read.csv("U:\\switched_proteomes\\total_proteome\\W-X\\W-F\\results\\step4\\REF_report.pr_matrix_with_matches.csv")
```


# for all dataframes, make a new column grouping gene with stripped sequence
                                   
## only need matches and gene_sequence combo from matches_W_X dataframe
```{r}
# from volcano plot of reference proteome, filter for genes of interest (with a switched peptide sequence)

gene_list <- pr_log2c_wt10g_ko10g %>% filter(Matches.Count > 0) %>% distinct(Genes) %>% pull(Genes)


# format data 


cols_to_pivot <- c("WT_UT_1","WT_UT_2","WT_UT_3",
                   "WT_10G_1","WT_10G_2","WT_10G_3",
                   "KO_UT_1","KO_UT_2","KO_UT_3",
                   "KO_10G_1","KO_10G_2","KO_10G_3")


ref$Matches.Count[ref$Matches.Count == "NaN"] <- 0

ref$Matches.Count <- sub("^([0-9.]+);.*", "\\1", ref$Matches.Count) 

ref <- ref %>% filter(Genes %in% gene_list) %>% filter(Matches.Count > 0) %>% 
  mutate(Gene_Sequence_Group = paste(Genes, Stripped.Sequence, sep = "_")) %>% 
    pivot_longer(cols = all_of(cols_to_pivot),
                                names_to="sample",
                                values_to="intensity")

ref <- ref %>% tidyr::extract(
    sample,
    into = c("condition", "replicate"),
    regex = "^(.*)_(\\d+)$"
  ) %>%  mutate(Gene_Sequence_Group = str_split(Gene_Sequence_Group, ";") %>% 
           map_chr(~ .x[1]))

ref_summary <- ref %>%
  group_by(Gene_Sequence_Group,condition) %>%
   summarise(
    mean_intensity = mean(intensity, na.rm = TRUE),
    sd_intensity   = sd(intensity, na.rm = TRUE),
    se_intensity   = sd(intensity, na.rm = TRUE) / sqrt(n()),
    .groups = "drop"
  )

ref$condition <- factor(ref$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))

ref_summary$condition <- factor(ref_summary$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))



```


# repeat for switched proteome results

## switch with matches data

```{r}
  switch <- read.csv("U:\\switched_proteomes\\total_proteome\\W-X\\W-F\\results\\step4\\SWITCH_report.pr_matrix_with_matches.csv")
```


# for all dataframes, make a new column grouping gene with stripped sequence
## only need matches and gene_sequence combo from matches_W_F dataframe
```{r}
# from volcano plot of reference proteome, filter for genes of interest
switch$Matches.Count[switch$Matches.Count == "NaN"] <- 0

switch$Matches.Count <- sub("^([0-9.]+);.*", "\\1", switch$Matches.Count) 

switch <- switch %>% filter(Genes %in% gene_list) %>% filter(Matches.Count > 0) %>%  
 mutate(Stripped.Sequence.Reference.Database = str_split(Stripped.Sequence.Reference, ";") %>% 
           map_chr(~ .x[1])) %>% 
  mutate(Gene_Sequence_Group = paste(Genes, Stripped.Sequence.Reference.Database, sep = "_")) %>% 
    pivot_longer(cols = all_of(cols_to_pivot),
                                names_to="sample",
                                values_to="intensity") %>%  tidyr::extract(
    sample,
    into = c("condition", "replicate"),
    regex = "^(.*)_(\\d+)$"
  )

switch_summary <- switch %>%
  group_by(Gene_Sequence_Group,condition) %>%
 summarise(
    mean_intensity = mean(intensity, na.rm = TRUE),
    sd_intensity   = sd(intensity, na.rm = TRUE),
    se_intensity   = sd(intensity, na.rm = TRUE) / sqrt(n()),
    .groups = "drop"
  )


switch$condition <- factor(switch$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))

switch_summary$condition <- factor(switch_summary$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))



```



# In order to combine REF and SWITCh data, we need to assign a common column between them (R won't recognise two peptides with just one sub e.g. ABCDEWFGHI and ABCDEFGHI)


```{r}
switch_bind <- switch %>% separate(Gene_Sequence_Group, into = c("Gene", "Sequence_Group"), sep = "_", remove = FALSE) %>%  select(Gene_Sequence_Group, Gene, condition, replicate, intensity) %>% mutate(proteome="switched")

ref_bind <- ref %>% separate(Gene_Sequence_Group, into = c("Gene", "Sequence_Group"), sep = "_", remove = FALSE) %>% 
  select(Gene_Sequence_Group, Gene, condition, replicate, intensity) %>% 
  mutate(proteome="reference") 
```

# join dataframes

```{r}
combined <- rbind(switch_bind,ref_bind)

combined <- combined %>% group_by(Gene_Sequence_Group, condition, replicate, proteome) %>% unique()
```

# normalise switches against reference proteome

```{r}
control_means <- combined %>% filter(proteome=="reference") %>% group_by (Gene_Sequence_Group, condition) %>% summarise(mean_intensity=mean(intensity))
```

# get abundance as proportion of reference proteome

```{r}
combined_merged <- combined %>%  filter(!proteome=="reference") %>% 
  left_join(control_means, by = c("Gene_Sequence_Group","condition"), relationship = "many-to-many") %>%
  mutate(intensity_against_ref = (intensity/mean_intensity)*100)
```

# plot intensities

```{r}

combined_summary <- combined_merged %>%
  group_by(Gene_Sequence_Group, condition, proteome) %>%
  summarise(
    mean_intensity = mean(intensity, na.rm = TRUE),
    sd_intensity   = sd(intensity, na.rm = TRUE),
    se_intensity   = sd(intensity, na.rm = TRUE) / sqrt(n()),
    .groups = "drop"
  )


combined_merged$condition <- factor(combined_merged$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))

combined_summary$condition <- factor(combined_summary$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))

legend_title <- "Proteome"

setwd("U:\\switched_proteomes\\total_proteome\\W-X\\W-F\\ALL_SWITCHES")

pdf("ALL_SWITCHES_NORMALISED.pdf")  # Open a PDF device

unique_combos <- unique(combined_merged$Gene_Sequence_Group)


for (combo in unique_combos) {
  sub_df <- combined_merged %>% filter(Gene_Sequence_Group == combo)
  
    sub_summary <- combined_summary %>% filter(Gene_Sequence_Group == combo)
  
  p <- ggplot() +
  geom_point(data = sub_df, aes(x = condition, y = intensity_against_ref, color = proteome),
             position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.75),
             alpha = 0.6) +  # Jitter to reduce overplotting
  
  geom_point(data = sub_summary, aes(x = condition, y = mean_intensity, color = proteome),
             position = position_dodge(width = 0.75), size = 3) +
  
  geom_errorbar(data = sub_summary, 
                aes(x = condition, ymin = mean_intensity - se_intensity, ymax = mean_intensity + se_intensity, color = proteome),
                position = position_dodge(width = 0.75), width = 0.2) +
  
  theme_minimal() +
  theme(axis.text = element_text(size = 10),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.position = "right",
        panel.border = element_rect(color = "gray3", fill = NA),
        text = element_text(size = 12, family = "Helvetica")) +
    scale_color_manual(legend_title,values = c("switched" = "#1c00fd"),
                       labels = c("W>D"))+
  xlab("") +
    scale_x_discrete(labels=c("WT_UT" = "Parental UT", "KO_UT" = "KO UT",
                              "WT_10G" = "Parental 10Gy",
                              "KO_10G" = "KO 10Gy"))+
  ylab("Intensity as percentage of reference proteome (%)") +
  ggtitle(combo)  # Optional: Add the group as title

  print(p)  # Print to PDF
}

dev.off()
```
# filter for anova significance

```{r}
anova_results <- list()
tukey_results <- list()

# Loop through each Gene_Position
for (gene in unique(combined_merged$Gene_Sequence_Group)) {
  for (prot in unique(combined_merged$proteome)) {
    sub_data <- combined_merged %>% 
      filter(Gene_Sequence_Group == gene, proteome == prot)

    # Check if at least two unique Conditions exist
    if (length(unique(sub_data$condition)) < 2) next

    # Fit ANOVA
    fit <- aov(intensity_against_ref ~ condition, data = sub_data)
    anova_res <- summary(fit)

    # Save ANOVA p-value
    pval <- anova_res[[1]][["Pr(>F)"]][1]

    anova_results[[paste(gene, prot, sep = "_")]] <- list(
      gene = gene,
      proteome = prot,
      p_value = pval
    )
  # Tukey's HSD if ANOVA is significant (e.g. p < 0.05)
    if (!is.na(pval) && pval < 0.05) {
      tukey <- TukeyHSD(fit)
      tukey_df <- as.data.frame(tukey$condition)
      tukey_df$comparison <- rownames(tukey_df)
      tukey_df$Gene_Position <- gene
      tukey_df$Proteome <- prot

      tukey_results[[paste(gene, prot, sep = "_")]] <- tukey_df
    }
  }
}

anova_df <- dplyr::bind_rows(anova_results, .id = "gene")
tukey_df <- bind_rows(tukey_results)

# hits
gene_list_signif <- anova_df %>% filter(p_value < 0.05) %>% distinct(gene) %>% pull(gene) 

gene_list_signif <- str_remove(gene_list_signif, "_switched$")

```
#produce dpe plot only highlighting switches with signif anova test

```{r}
plot <- pr_log2c_wt10g_ko10g  %>% 
  mutate(Gene_Sequence_Group = paste(Genes, Stripped.Sequence, sep = "_"))  

plot <- plot %>% ggplot(aes(x=Difference,y=X.NAME.)) +
  geom_point()+ geom_hline(yintercept = 1.3, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed") + geom_vline(xintercept = 1, linetype="dashed")+
    gghighlight(Matches.Count > 0, Gene_Sequence_Group %in% gene_list_signif) +
  geom_text(
    data = plot %>% filter(Matches.Count > 0) %>% filter(Gene_Sequence_Group %in% gene_list_signif),
    aes(label = Genes),
    position = position_jitter(width = 0.2, height = 0.25),
    color = "blue")+ 
    theme(axis.text = element_text(size = 10),
        plot.title = element_text(hjust = 0.5),
       legend.position = "none",
       panel.border = element_rect(color="gray3", fill= NA),
       panel.background=element_blank(),
        text=element_text(size = 12,family = "Helvetica"))+ xlab("log2 fold change")+
  ylab("-log p") +ggtitle("W>F peptides present in WT 10Gy versus KO 10Gy")
```

# produce plot with only significant hits

```{r}


combined_merged <- combined %>%  filter(!proteome=="reference") %>% filter(Gene_Sequence_Group %in% gene_list_signif) %>% 
  left_join(control_means, by = c("Gene_Sequence_Group","condition"), relationship = "many-to-many") %>%
  mutate(intensity_against_ref = (intensity/mean_intensity)*100)

combined_summary <- combined_merged %>%
  group_by(Gene_Sequence_Group, condition, proteome) %>%
   summarise(
    mean_intensity = mean(intensity_against_ref, na.rm = TRUE),
    sd_intensity   = sd(intensity_against_ref, na.rm = TRUE),
    se_intensity   = sd(intensity_against_ref, na.rm = TRUE) / sqrt(n()),
    .groups = "drop"
  )

combined_merged$condition <- factor(combined_merged$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))

combined_summary$condition <- factor(combined_summary$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))

legend_title <- "Proteome"

setwd("U:\\switched_proteomes\\total_proteome\\W-X\\W-F\\ALL_SWITCHES")

pdf("ALL_SWITCHES_NORMALISED.pdf", width = 3, height = 5)  # Open a PDF device

unique_combos <- unique(combined_merged$Gene_Sequence_Group)


for (combo in unique_combos) {
  sub_df <- combined_merged %>% filter(Gene_Sequence_Group == combo)
  
    sub_summary <- combined_summary %>% filter(Gene_Sequence_Group == combo)
  
  p <- ggplot() +
  geom_point(data = sub_df, aes(x = condition, y = intensity_against_ref, color = proteome),
             position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.75),
             alpha = 0.6) +  # Jitter to reduce overplotting
  
  geom_point(data = sub_summary, aes(x = condition, y = mean_intensity, color = proteome),
             position = position_dodge(width = 0.75), size = 3) +
  
  geom_errorbar(data = sub_summary, 
                aes(x = condition, ymin = mean_intensity - se_intensity, ymax = mean_intensity + se_intensity, color = proteome),
                position = position_dodge(width = 0.75), width = 0.2) +
  
  theme_minimal() +
  theme(axis.text = element_text(size = 10),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.position = "right",
        panel.border = element_rect(color = "gray3", fill = NA),
        text = element_text(size = 12, family = "Helvetica")) +
    scale_color_manual(legend_title,values = c("switched" = "#a70d56"),
                       labels = c("K>M"))+
  xlab("") +
    scale_x_discrete(labels=c("WT_UT" = "Parental UT", "KO_UT" = "KO UT",
                              "WT_10G" = "Parental 10Gy",
                              "KO_10G" = "KO 10Gy"))+
  ylab("Intensity as percentage of reference proteome (%)") +
  ggtitle(combo)  # Optional: Add the group as title

  print(p)  # Print to PDF
}

dev.off()
```

# Export all results for X>X switches (e.g. W>F)

```{r}
write.csv(anova_df, file="U:\\switched_proteomes\\total_proteome\\W-X\\WF\\ALL_SWITCHES\\anova_res.csv", row.names = FALSE)
write.csv(combined, file="U:\\switched_proteomes\\total_proteome\\W-X\\WF\\ALL_SWITCHES\\anova_res.csv", row.names = FALSE))
write.csv(combined_merged, file="U:\\switched_proteomes\\total_proteome\\W-X\\WF\\ALL_SWITCHES\\anova_res.csv", row.names = FALSE))
write.csv(combined_summary, file="U:\\switched_proteomes\\total_proteome\\W-X\\WF\\ALL_SWITCHES\\anova_res.csv", row.names = FALSE))
write.csv(tukey_df, file="U:\\switched_proteomes\\total_proteome\\W-X\\WF\\ALL_SWITCHES\\anova_res.csv", row.names = FALSE))
ggsave(plot,file="U:\\switched_proteomes\\total_proteome\\W-X\\WF\\ALL_SWITCHES\\plot.pdf", width = 7, height=7, scale=1, device=cairo_pdf)
```
# Repeat for all switches of interest 

# Load all data for all X>X switches (e.g to load all W>X switches data)
```{r}

root_dir <- "U:\\switched_proteomes\\total_proteome\\W-X"


proteome_dirs <- list.dirs(root_dir, recursive = FALSE, full.names = TRUE)
proteome_dirs <- proteome_dirs[grepl("W", basename(proteome_dirs))]

file_paths <- file.path(proteome_dirs, "ALL_SWITCHES", "SWITCH_INTENSITY_NORMALISED.csv")


file_paths <- file_paths[file.exists(file_paths)]
if(length(file_paths) == 0) stop("No CSV files found!")


data_list <- lapply(file_paths, function(fp) {
  df <- read.csv(fp, stringsAsFactors = FALSE)
  
  
  if(nrow(df) == 0) return(NULL)
  
  
  df$proteome <- basename(dirname(dirname(fp)))
  
  df
})

data_list <- Filter(Negate(is.null), data_list)

combined_df <- bind_rows(data_list)

```
# Load all summary data

```{r}

root_dir <- "U:\\switched_proteomes\\total_proteome\\W-X"


proteome_dirs <- list.dirs(root_dir, recursive = FALSE, full.names = TRUE)
proteome_dirs <- proteome_dirs[grepl("W", basename(proteome_dirs))]

file_paths <- file.path(proteome_dirs, "ALL_SWITCHES", "SWITCH_INTENSITY_NORMALISED_SUMMARY.csv")


file_paths <- file_paths[file.exists(file_paths)]
if(length(file_paths) == 0) stop("No CSV files found!")


data_list <- lapply(file_paths, function(fp) {
  df <- read.csv(fp, stringsAsFactors = FALSE)
  
  
  if(nrow(df) == 0) return(NULL)
  
  
  df$proteome <- basename(dirname(dirname(fp)))
  
  df
})

data_list <- Filter(Negate(is.null), data_list)

combined_df_summary <- bind_rows(data_list)

```

# Load all ANOVA data

```{r}
root_dir <- "U:\\switched_proteomes\\total_proteome\\W-X"


proteome_dirs <- list.dirs(root_dir, recursive = FALSE, full.names = TRUE)
proteome_dirs <- proteome_dirs[grepl("W", basename(proteome_dirs))]

file_paths <- file.path(proteome_dirs, "ALL_SWITCHES", "anova_res.csv")


file_paths <- file_paths[file.exists(file_paths)]
if(length(file_paths) == 0) stop("No CSV files found!")


data_list <- lapply(file_paths, function(fp) {
  df <- read.csv(fp, stringsAsFactors = FALSE)
  
  
  if(nrow(df) == 0) return(NULL)
  
  
  df$proteome <- basename(dirname(dirname(fp)))
  
  df
})

data_list <- Filter(Negate(is.null), data_list)

combined_df_anova <- bind_rows(data_list)
```

# Load all Tukey HSD data

```{r}
root_dir <- "U:\\switched_proteomes\\total_proteome\\W-X"


proteome_dirs <- list.dirs(root_dir, recursive = FALSE, full.names = TRUE)
proteome_dirs <- proteome_dirs[grepl("W", basename(proteome_dirs))]

file_paths <- file.path(proteome_dirs, "ALL_SWITCHES", "tukey_res.csv")


file_paths <- file_paths[file.exists(file_paths)]
if(length(file_paths) == 0) stop("No CSV files found!")


data_list <- lapply(file_paths, function(fp) {
  df <- read.csv(fp, stringsAsFactors = FALSE)
  
  
  if(nrow(df) == 0) return(NULL)
  
  
  df$proteome <- basename(dirname(dirname(fp)))
  
  df
})

data_list <- Filter(Negate(is.null), data_list)

combined_df_tukey <- bind_rows(data_list)
```

# Filter combined_df and combined_df_summary by only those switched peptides that exhibit significance (p < 0.05) in ANVOA test

```{r}
gene_list_signif <- combined_df_anova %>% filter(p_value <= 0.05)
gene_list_signif <- str_remove(gene_list_signif$gene, "_switched$")
gene_list_signif <- as.data.frame(gene_list_signif)

combined_df_filtered <- combined_df %>% filter(Gene_Sequence_Group %in% gene_list_signif$gene)

combined_df_summary_filtered <- combined_df_summary %>% filter(Gene_Sequence_Group %in% gene_list_signif$gene)
```

# Build a plot to show all switches for a given gene

```{r}
cols <- c("W-C" = "#32ff03",
  "W-T" = "#aff480",
  "W-Q" = "#d269ff",
  "W-R" = "#fb91e1",
  "W-M" = "#a70d56",
  "W-H" = "#F2E500",
  "W-A" = "#ff1616",
  "W-D" = "#1c00fd",
  "W-E" = "#ffd0d4",
  "W-F" = "#ff9135",
  "W-G" = "#0dd1fd",
  "W-I" = "#16752a",
  "W-K" = "#ff00de",
  "W-L" = "#623282",
  "W-N" = "#0dfdd8",
  "W-P" = "#704d00",
  "W-S" = "#6e949c",
  "W-V" = "#7f96f9",
  "W-Y" = "#ff1c80"
)


combined_df_filtered$condition <- factor(combined_df_filtered$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))

combined_df_summary_filtered$condition <- factor(combined_df_summary_filtered$condition, levels = c("WT_UT",
                                                                "KO_UT",
                                                                "WT_10G",
                                                                "KO_10G"))

setwd("U:\\switched_proteomes\\total_proteome\\W-X\\ALL_SWITCHES_NORMALISED")

pdf("NORMALISED_SWITCHES.pdf")  # Open a PDF device

unique_combos <- unique(combined_df_filtered$Gene_Sequence_Group)


for (combo in unique_combos) {
  sub_df <- combined_df_filtered %>% filter(Gene_Sequence_Group == combo)
  
    sub_summary <- combined_df_summary_filtered %>% filter(Gene_Sequence_Group == combo)
  
  p <- ggplot() +
  geom_point(data = sub_df, aes(x = interaction(condition,proteome), y = intensity_against_ref, color = proteome),
             position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.75),
             alpha = 0.6) +  # Jitter to reduce overplotting
  
  geom_point(data = sub_summary, aes(x = interaction(condition,proteome), y = mean_intensity, color = proteome),
             position = position_dodge(width = 0.75), size = 3) +
  
  geom_errorbar(data = sub_summary, 
                aes(x = interaction(condition,proteome), ymin = mean_intensity - se_intensity, ymax = mean_intensity + se_intensity, color = proteome),
                position = position_dodge(width = 0.75), width = 0.2) +
    scale_color_manual(values=cols)+theme_minimal() +
  theme(axis.text = element_text(size = 10),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.position = "right",
        panel.border = element_rect(color = "gray3", fill = NA),
        text = element_text(size = 12, family = "Helvetica")) +
  xlab("") +
    scale_x_discrete(labels=c("WT_UT" = "Parental UT", "KO_UT" = "KO UT",
                              "WT_10G" = "Parental 10Gy",
                              "KO_10G" = "KO 10Gy"))+
  ylab("Intensity of switched peptide as percentage of reference (%)") +
  ggtitle(combo) 

  print(p)  # Print to PDF
}

dev.off()
```
# Save data

```{r}
write.csv(combined_df, file="U:\\switched_proteomes\\total_proteome\\W-X\\ALL_SWITCHES_NORMALISED\\combined_df.csv", row.names = FALSE)
write.csv(combined_df_anova, file="U:\\switched_proteomes\\total_proteome\\W-X\\ALL_SWITCHES_NORMALISED\\combined_df.csv", row.names = FALSE)
write.csv(combined_df_summary, file="U:\\switched_proteomes\\total_proteome\\W-X\\ALL_SWITCHES_NORMALISED\\combined_df.csv", row.names = FALSE)
write.csv(combined_df_tukey, file="U:\\switched_proteomes\\total_proteome\\W-X\\ALL_SWITCHES_NORMALISED\\combined_df.csv", row.names = FALSE)



```

# Load Tukey results - in this example, I am only interested in mistranslated peptides that are significantly upregulated WT_10G vs KO_10G, WT_10G vs KO_UT, WT_10G vs WT_UT and not significantly different in WT_UT vs KO_UT, KO_UT vs KO_10G


```{r}

W_X_stats <- read.csv("U:\\switched_proteomes\\total_proteome\\W-X\\ALL_SWITCHES_NORMALISED\\combined_df_tukey.csv")

W_X_stats <- W_X_stats %>% mutate(Proteome="W>X")


# List of peptides that have significantly adjusted p values in conditions of interest
W_X_stats_filtered <- W_X_stats %>%
  group_by(Gene_Position, proteome) %>%
  summarise(
    sig_WT10G_WTUT  = any(comparison == "WT_10G-WT_UT"  & p.adj <= 0.05),
    sig_WT10G_KOUT  = any(comparison == "WT_10G-KO_UT"  & p.adj <= 0.05),
    sig_WT10G_KO10G = any(comparison == "KO_10G-WT_10G" & p.adj <= 0.05),
    
    ns_KOUT_KO10G   = any(comparison == "KO_10G-KO_UT"  & p.adj >  0.05),
    ns_WTUT_KOUT    = any(comparison == "KO_UT-WT_UT"  & p.adj >  0.05),
    
    .groups = "drop"
  ) %>%
  filter(
    sig_WT10G_WTUT,
    sig_WT10G_KOUT,
    sig_WT10G_KO10G,
    ns_KOUT_KO10G,
    ns_WTUT_KOUT
  )

# Filter raw statistical values

combined_stats <- combined_stats %>% filter(Gene_Position %in% combined_stats_filtered$Gene_Position)

```


